shared/middleware/files_api.rb

```diff
@@ -44,6 +44,12 @@ def can_view_abusive_assets?(encrypted_channel_id)
     teaches_student?(owner_user_id)
   end
 
+  def codeprojects_can_view?(encrypted_channel_id)
+    owner_storage_id, _ = storage_decrypt_channel_id(encrypted_channel_id)
+    owner_user_id = user_storage_ids_table.where(id: owner_storage_id).first[:user_id]
```

Comment:
need to rescue the same 2 exceptions here. It might also be worth extracting these two lines (plus the rescue) into a helper since they are used twice.